UIE-92 Decouple Nav Router - Environments Page

msilva-broad opened the pull request:
To support Cross-team Code sharing effort...
- preliminary file move and splitting out of EnvironmentsPage wrapper from Environments component.  This sets us up for followup cleanup that will further decouple the nav/router dependency.
- Environments component in src/analysis area now promoted to a subfolder in preperation for further code breakup (it has a few subcomponents that should be broken out from the file soon)
- added unit test coverage for EnvironmentsPage wrapper which now lives back in src/pages area.

### Jira Ticket: [](https://broadworkbench.atlassian.net/browse/UIE-92)


### Testing strategy
no functional change

msilva-broad committed:
UIE-92 Decouple Nav Router
- preliminary file move and splitting out of EnvironmentsPage wrapper from Environments component.  This sets us up for followup cleanup that will further decouple the nav/router dependency.
- Environments component in src/analysis area now promoted to a subfolder in preperation for further code breakup (it has a few subcomponents that should be broken out from the file soon)
- added unit test coverage for EnvironmentsPage wrapper which now lives back in src/pages area.

nawatts commented on src/libs/routes.js:
```diff
@@ -3,7 +3,7 @@ import { compile, pathToRegexp } from 'path-to-regexp';
 import * as AnalysisLauncher from 'src/analysis/AnalysisLauncher';
 import * as AppLauncher from 'src/analysis/AppLauncher';
-import * as Environments from 'src/analysis/Environments';
+import * as Environments from 'src/analysis/Environments/Environments';
```
Environments no longer defines navPaths, so the app fails to start with this. The routes module needs to import EnvironmentsPage instead of Environments.

msilva-broad replied:
oooh good  catch.  I had eyes on this and STILL missed it... odd that I did an app launch test at some point and this slipped thru.

nawatts replied:
The e2e tests caught it 😁 
https://app.circleci.com/pipelines/github/DataBiosphere/terra-ui/17929/workflows/89ea8361-abbb-47b3-8a8a-ed6a997480b5/jobs/66352/tests

msilva-broad replied:
fixed

msilva-broad committed:
UIE-92 Decouple Nav Router
- fixed routes mapping bug that somehow slipped past my testing
- also exported a type from AzureStorage.ts to silence an error in ajax.js root file complaining about reliance on private type.

nawatts commented on src/analysis/Environments/Environments.ts:
```diff
@@ -248,10 +245,11 @@ type AppWithWorkspace = DecoratedResourceAttributes & App;
 }
-export const Environments = ({ nav = undefined }: EnvironmentsProps) => {
+
+export const Environments: React.FC<EnvironmentsProps> = ({ nav = undefined }) => {
```
A lot of code uses the existing form of type definitions for function components. I think adding React.FC would be something of a break with convention.

nawatts replied:
https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components has a comparison of different styles of type declarations as well as some issues with `React.FC`.

msilva-broad replied:
Sorry I have not been better about surfacing this recommendation.  I updated the TS tips doc... `React.FC<PropsType> `has some nice benefits and overall reduces the signature burden while keeping the desired type safety.

nawatts replied:
It doesn't matter so much now that we're on React 18, but based on the linked create-react-app issue and the incompatibility with generics that it mentions, I think I'd disagree with that recommendation.

nawatts commented on src/libs/routes.js:
```diff
@@ -3,10 +3,10 @@ import { compile, pathToRegexp } from 'path-to-regexp';
 import { routeHandlersStore } from 'src/libs/state';
 import * as AzurePreview from 'src/pages/AzurePreview';
 import * as Projects from 'src/pages/billing/List/List';
+import * as Environments from 'src/pages/EnvironmentsPage';
```
Nit: It would be nice to match the variable name to the module (EnvironmentsPage instead of Environments).

jdcanas commented on src/libs/ajax/AzureStorage.ts:
```diff
@@ -19,7 +19,7 @@ type StorageDetails = {
 };
 
-interface AzureFileRaw {
+export interface AzureFileRaw {
```
this shouldn't need to be exported

msilva-broad replied:
actually it did to silence that TS error/warning in my IDE (IntelliJ)... it was complaining in ajax.js root file that something being used there was depending on this secondary type.

nawatts replied:
I think a better way to get rid of that warning is to change this to a `type` instead of an `interface`.
